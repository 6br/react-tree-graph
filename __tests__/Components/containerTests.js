import React from 'react';
import { shallow } from 'enzyme';

import Container from '../../Components/container';

const nodes = [
	{
		x: 1,
		y: 2,
		data: {
			name: 'Colour'
		}
	}, {
		x: 100,
		y: 50,
		data: {
			name: 'Black'
		}
	}
];

describe('<Container>', () => {
	test('renders correctly', () => {
		const props = {
			nodes: nodes,
			links: [{
				source: nodes[0],
				target: nodes[1]
			}],
			height: 100,
			keyProp: 'name',
			labelProp: 'name',
			nodeOffset: 3.5,
			nodeRadius: 5,
			width: 200,
			circleProps: {},
			gProps: {},
			pathProps: {},
			svgProps: {},
			textProps: {}
		};

		const tree = shallow(<Container {...props}/>);

		expect(tree).toMatchSnapshot();
	});

	test('html tree props added', () => {
		const props = {
			nodes: [],
			links: [],
			height: 100,
			keyProp: 'name',
			labelProp: 'name',
			nodeOffset: 3.5,
			nodeRadius: 5,
			width: 200,
			circleProps: {},
			gProps: {},
			pathProps: {},
			svgProps: {
				className: 'test-class',
				stoke: 'none'
			},
			textProps: {}
		};

		const tree = shallow(<Container {...props}/>);

		expect(tree).toMatchSnapshot();
	});
});