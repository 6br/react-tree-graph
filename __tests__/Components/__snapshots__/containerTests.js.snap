// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container> html tree props added 1`] = `
<svg
  className="test-class"
  height={100}
  stoke="none"
  width={200}
/>
`;

exports[`<Container> renders correctly 1`] = `
<svg
  height={100}
  width={200}
>
  <Link
    key="Black"
    keyProp="name"
    pathProps={Object {}}
    source={
      Object {
        "data": Object {
          "name": "Colour",
        },
        "x": 1,
        "y": 2,
      }
    }
    target={
      Object {
        "data": Object {
          "name": "Black",
        },
        "x": 100,
        "y": 50,
      }
    }
    x1={1}
    x2={100}
    y1={2}
    y2={50}
  />
  <Node
    circleProps={Object {}}
    gProps={Object {}}
    key="Colour"
    keyProp="name"
    labelProp="name"
    name="Colour"
    offset={3.5}
    radius={5}
    textProps={Object {}}
    x={1}
    y={2}
  />
  <Node
    circleProps={Object {}}
    gProps={Object {}}
    key="Black"
    keyProp="name"
    labelProp="name"
    name="Black"
    offset={3.5}
    radius={5}
    textProps={Object {}}
    x={100}
    y={50}
  />
</svg>
`;
